### PR TITLE
Project 4 spec

### PR DESCRIPTION
**Pull request by dylanvu**
image

Check `project_4.html` for the page

### CODE REVIEW

**Review by bcbielecki — CHANGES_REQUESTED** [review]
Let's hold off on closing this PR just until the Project 1 PR responsiveness fixes go through and updates this one. When it does, please mark the change request as fixed. Awesome work!

File: project_4.html [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+						<ul class="nav">
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-arduino-nano">Arduino Nano</a>
+							</li>
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-pinout">Pinout</a>
+							</li>
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-potentiometer">Potentiometer</a>
+								<ul class="nav">
```
Comment (bcbielecki) [inline project_4.html:61]: Please remove the subheadings below "Potentiometer" from the sidebar. It clutters the space with less important headings.

File: project_4.html [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+									<li class="nav-item">
+										<a class="nav-link" href="#pj-heading-using-it-with-arduino-potentiometer">Using it with Arduino</a>
+									</li>
+								</ul>
+							</li>
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-rgb-led">RGB LED</a>
+								<ul class="nav">
+									<li>
+										<a class="nav-link" href="#pj-heading-using-it-with-arduino-rgb">Using it with Arduino</a>
```
Comment (bcbielecki) [inline project_4.html:74]: Please remove the subheadings below "RGB LED" from the sidebar. It clutters the space with less important headings.

File: project_4.html [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-arduino-ide">Arduino IDE</a>
+							</li>
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-ch340-driver">CH340 Driver</a>
+							</li>
+						</ul>
+					</li>
+					<li class="nav-item">
+						<a class="nav-link" href="#pj-heading-requirements">Requirements</a>
```
Comment (bcbielecki) [inline project_4.html:92]: Please add the three subheadings for "Requirements" to the sidebar to stay consistent with the other projects pages.

File: project_4.html [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+			<ol class="mb-5 list-group list-group-numbered">
+				<li class="list-group-item">
+					Build the circuit from <strong>Schematic A</strong> (LED Blinker) on your breadboard. You will design a blinking LED using the Arduino Nano.
+					<div class="alert alert-info mt-2" role="alert">
+						Use the pseudo code below to program an LED that blinks on and off.
+					</div>
+					<div class="alert alert-info mt-2" role="alert">
+						These functions may be helpful: <code class="pj-code language-arduino">Serial.begin()</code>, <code class="pj-code language-arduino">pinMode()</code>, <code class="pj-code language-arduino">digitalWrite()</code>, <code class="pj-code language-arduino">delay()</code>
+					</div>
+					<pre class="mt-3"><code class="pj-code language-plaintext">// Assign variable to pin number for LED
```
Comment (bcbielecki) [inline project_4.html:440]: The psuedo code is not color highlighted correctly. Please make sure this class is set to language-arduino. The same applies to all other code snippets.

File: project_4.html [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-ch340-driver">CH340 Driver</a>
+							</li>
+						</ul>
+					</li>
+					<li class="nav-item">
+						<a class="nav-link" href="#pj-heading-requirements">Requirements</a>
+					</li>
+					<li class="nav-item">
+						<a class="nav-link" href="#pj-heading-parts">Parts</a>
```
Comment (bcbielecki) [inline project_4.html:95]: Please add images for all of the non-resistor parts here.

Comment (bcbielecki) [reply project_4.html:95]: Please also append the arduino rev3 datasheet here. (Upload the pdf to the repo; please don't use a hyperlink to another site )

File: project_4.html [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+					<tr>
+						<td>Mini USB Cable</td>
+						<td>1</td>
+					</tr>
+				</tbody>
+			</table>
+			<!-- End Part Table -->
+
+
+			<h2 class="mb-3 fw-bold" id="pj-heading-schematics">Schematics</h2>
```
Comment (bcbielecki) [inline project_4.html:380]: Please reduce the max-width of the images for Schematics A and B to be more consistent with the part sizing of Schematic C

**Review by bcbielecki — CHANGES_REQUESTED** [review]
Hi! Here is the first wave of review comments.

And a couple general notes:

- [ ] Please update this PR with main ASAP, before you make any more changes. It is severely out-of-date. 
- [ ] Please change the roboto font cdn to use the same link as the other projects pages. It's currently affecting the font weights on your page.
- [ ] When you have completed all of these tasks, please re-request a review from me

File: project_4.html (hunk 5 above)
Comment (gavinn2) [reply project_4.html:95]: > Please also append the arduino rev3 datasheet here. (Upload the pdf to the repo; please don't use a hyperlink to another site )

Please clarify this, I wouldn't know how to do it without a hyperlink to the pdf? also this is the pdf I found: https://docs.arduino.cc/resources/datasheets/A000005-datasheet.pdf

File: project_4.html [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+						</td>
+						<td>1</td>
+					</tr>
+				</tbody>
+			</table>
+			<!-- End Part Table -->
+
+
+			<h2 class="mb-3 fw-bold" id="pj-heading-schematics">Schematics</h2>
+			<div class="accordion mb-5" id="schemAccordian">
```
Comment (bcbielecki) [inline project_4.html:401]: please check the opening and closing <\div\> tags here. They're not balanced.

File: project_4.html [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+						</button>
+					</h2>
+					<!--Pin to page?-->
+					<div id="schemCollapse3" class="accordion-collapse collapse" data-bs-parent="#schemAccordian">
+						<div class="accordion-body text-center">
+							<img src='./assets/images/project_4/ops_schem_proj4c.png' class="pj-schem" style="max-height: 600px; max-width: 600px;">
+						</div>
+					</div>
+				</div>
+				<div class = "mb-3">
```
Comment (bcbielecki) [inline project_4.html:446]: I think this code is extraneous

File: project_4.html [hunk 9]
```diff
@@ -0,0 +1,545 @@
...
+			<p>
+				Visit this <a href="https://learn.sparkfun.com/tutorials/how-to-install-ch340-drivers/all" target="_blank" rel="noopener noreferrer">link</a> to download the installer for the CH340 driver. You may have to scroll down the page a little. Make sure
+				to select the installer appropriate for your system.
+			</p>
+			<div class="mb-5" style= "text-align: center;">
+				<img class="img-fluid my-4" style="max-width: 625px;" src=".\assets\images\project_4\ops_install_ch340.png">
+			</div>
+
+
+			<h2 class="mb-3 fw-bolder" id="pj-heading-requirements">Requirements</h2>
```
Comment (bcbielecki) [inline project_4.html:291]: Please change these lists to bootstrap-styled lists

**Review by bcbielecki — CHANGES_REQUESTED** [review]
Keep up the good work! Here are some more review comments.

I also noticed that the content extends to the right beyond the viewport when the viewport width decreases. We should figure out what is causing it.

File: project_4.html [hunk 10]
```diff
@@ -0,0 +1,545 @@
...
+					<strong>Compress</strong> the folder to a <strong>zip</strong> file and rename the file using the format “<strong>ops_project#_lastname_firstname.zip</strong>” Then, submit the <strong>zip</strong> file to the Project 4 Canvas assignment.
+			</ul>
+		</main>
+	</div>
+	<!-- ======= Footer ======= -->
+	<script src="./js/ops_footer.js"></script>
+	<!-- End Footer -->
+</body>
+
+</html>
```
Comment (bcbielecki) [inline project_4.html:545]: We have a bug here. This <\pre\> tag needs to have a set width, so it doesn't push the other content off the right side of the viewport. Would you turn this into an issue?

For now replace the opening tag with <\pre style="white-space: pre-wrap;"\>

Do the same for the one above.

**Review by bcbielecki — CHANGES_REQUESTED** [review]
Keep up the good work! Here are some more review comments.

I also noticed that the content extends to the right beyond the viewport when the viewport width decreases. We should figure out what is causing it.

File: project_4.html [hunk 11]
```diff
@@ -0,0 +1,545 @@
...
+						<a class="nav-link" href="#pj-heading-overview">Overview</a>
+					</li>
+					<li class="nav-item">
+						<a class="nav-link" href="#pj-heading-concepts">Concepts</a>
+						<ul class="nav">
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-arduino-nano">Arduino Nano</a>
+							</li>
+							<li class="nav-item">
+								<a class="nav-link" href="#pj-heading-pinout">Pinout</a>
```
[the remark this answers is not in the document]
Comment (dylanvu) [reply project_4.html:57]: addressed

File: project_4.html [hunk 12]
```diff
@@ -0,0 +1,545 @@
...
+<head>
+    <meta charset="utf-8" name="viewport" content="width=device-width, initial-scale=1">
+    <!--Title-->
+    <title>Open Project Space</title>
+    <!--Favicon-->
+    <link rel="icon" href="./assets/images/ops_logo_favicon.png">
+    <!--Roboto Font CDN-->
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link
+        href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,700;1,100;1,300;1,700&display=swap"
```
[the remark this answers is not in the document]
Comment (dylanvu) [reply project_4.html]: done

File: project_4.html [hunk 13]
```diff
@@ -0,0 +1,545 @@
...
+    <!-- Highlight JS v11.7.0-->
+    <link rel="stylesheet" href="./css/highlight.js/default.min.css" type="text/css">
+    <link rel="stylesheet" href="./css/highlight.js/arduino-light.min.css" type="text/css">
+    <script src="./js/highlight.min.js"></script>
+    <script>hljs.highlightAll();</script>
+</head>
+
+<body style="">
+	<!-- ======= Header ======= -->
+	<script src="./js/ops_header.js"></script>
```
[the remark this answers is not in the document]
Comment (dylanvu) [reply project_4.html:38]: done

File: project_4.html (hunk 7 above)
Comment (dylanvu) [reply project_4.html:447]: fixed, I think

File: project_4.html (hunk 5 above)
Comment (dylanvu) [reply project_4.html:90]: done

**Review by bcbielecki — APPROVED** [review]
10/10

**Pull request merged**